eng/Publishing.props

```diff
@@ -19,7 +19,7 @@
       We only want to publish the FSharp assets externally when we're publishing all artifacts, not only RID-specific ones.
       Add these packages as "Vertical" visibility when we're only publishing RID-specific packages so we don't publish this one.
     -->
-    <FSharpArtifactVisibility Condition="'$(EnableDefaultRidSpecificArtifacts)' == 'true'">Vertical</FSharpArtifactVisibility>
+    <FSharpArtifactVisibility Condition="'$(EnableDefaultRidSpecificArtifacts)' == 'true' or '$(DotNetBuildPass)' != '1'">Vertical</FSharpArtifactVisibility>
```

Comment:
@jkoritzinsky While this should work, I kind of suspect this should be moved up to https://github.com/dotnet/sdk/pull/47914/files#diff-350198cd1e3783d062a0d2dcee549e02f83dae123340fa4be238637ee421b527R15, **or** EnableDefaultArtifacts shouljd not be set for BP2 in sdk.

Thoughts?

/cc @ViktorHofer

Reply:
EnableDefaultArtifacts being set makes sense for BP2 as it reduces the burden of including artifacts manually (especially for the cross-DAC packages, where RID-specific inclusion won't capture the packages). I'm fine with a fix like this.

Reply:
```suggestion
    <FSharpArtifactVisibility Condition="'$(EnableDefaultRidSpecificArtifacts)' == 'true' or ('$(DotNetBuildPass)' != '' and '$(DotNetBuildPass)' != '1')">Vertical</FSharpArtifactVisibility>
```